src/xcode/ENA/ENA/Source/Scenes/CreateAntigenTestProfile/CreateAntigenTestProfileViewController.swift

```diff
@@ -189,7 +189,10 @@ class CreateAntigenTestProfileViewController: UITableViewController, FooterViewH
 	
 	@objc
 	private func dateOfBirthDidChange(datePicker: UIDatePicker) {
-		viewModel.update(datePicker.date, keyPath: \.dateOfBirth)
+		let offset = datePicker.timeZone?.secondsFromGMT() ?? 0
+		let utcDate = datePicker.date
+		let date = utcDate.addingTimeInterval(TimeInterval(offset))
+		viewModel.update(date, keyPath: \.dateOfBirth)
```

Comment:
Wouldn't it be possible to let the user pick the time in UTC instead of converting it to the equivalent UTC date manually?

Reply:
And I just checked another implementation detail: we are currently using UTC to decode the date without time information, so it is decoded as the UTC midnight date. But then we are using a formatter with local time zone in the `AntigenTestProfileViewModel`. I think we should also use a etc date formatter here, because the profile breaks if it is viewed in time zones that have an offset < 0 to UTC. We're just lucky that most of our users are in a time zone where it works.

Reply:
@nickguendling The user is picking the that in UTC. That is exactly the issue here. That is why I converted the date to local time because when encoding it is converted to just date.

Reply:
The default time zone for the date picker is the local time so it interprets the selected date in the local time zone. `ISO8601DateFormatter.justDate` uses UTC to interpret that date. It is an option to now calculate the UTC date that has the same date components as the selected local date has (although I would suggest a different naming since the `utcDate` is the selected date in local time, not UTC). Or you can just set the UIDatePicker to UTC to interpret it in UTC already (I tried it 😉). And since or en- and decoding uses UTC, you need to use a UTC date formatter to show it.

Reply:
‎

Reply:
I do not understand the discussion currently. This PR fixes the issue. Please open another feature ticket if you want to do a code refactoring.

Reply:
@nickguendling Please compare the following two screenshots. Check the time on my device, the date I selected in the app and the debug output from the date picker's date. How is that not UTC?

![Bildschirmfoto 2021-05-19 um 15 44 01](https://user-images.githubusercontent.com/3601228/118823598-62253c00-b8b9-11eb-9c94-298dbabfafbd.png)
![IMG_2970](https://user-images.githubusercontent.com/3601228/118823603-63566900-b8b9-11eb-92e0-a47a85c195e3.PNG)

Reply:
Since a `Date` object does not contain time zone information, it is shown in its UTC representation by the debugger. The date only contains the seconds from the reference date (2001-01-01 00:00 UTC) to the date selected by the user (2020-05-19 00:42:59 UTC+02:00). So it can only be interpreted correctly as the date of birth if you know the time zone it was selected in. That is the bug. We interpret it in UTC, while we have to take the user time zone into account to interpret it correctly. That's what your solution does: It calculates the point in time when UTC has the same time (00:42:59) as the user had locally when selecting the date. After that calculation, it can be correctly interpreted in the UTC time zone. So if you want to call something the `utcDate`, it would be the result of the calculation, not the input. Because date calculations and time zones are hard and we can make it simpler: start with UTC in the date picker, then the calculation is not needed.

Reply:
I understand all that but you literally said - and I quote - "The naming of utcDate for a local date is misleading[...]". But this is a UTC date. So why is naming the UTC date "utcDate" wrong?